### PR TITLE
[libpqxx] update to 7.7.3

### LINKED ISSUE #24311: [libpqxx] update to 7.7.3
Library name: libpqxx

New version number: 7.7.3

https://github.com/jtv/libpqxx/releases/tag/7.7.3

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #24311 

Update [libpqxx] versionto 7.7.3

Feature test successfully in the following triplet:

- x86-windows
- x64-windows
- x64-windows-static

### CODE REVIEW

File: ports/libpqxx/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "libpqxx",
-  "version-semver": "7.6.0",
-  "port-version": 3,
+  "version-semver": "7.7.3",
   "description": "The official C++ client API for PostgreSQL",
   "homepage": "https://www.postgresql.org/",
   "dependencies": [
```
Comment (Adela0814) [inline ports/libpqxx/vcpkg.json:7]: please add license

File: ports/libpqxx/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "libpqxx",
-  "version-semver": "7.6.0",
-  "port-version": 3,
+  "version-semver": "7.7.3",
```
Comment (Adela0814) [inline ports/libpqxx/vcpkg.json:3]: version

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/libpqxx/fix_build_with_apple_clang_13.patch [hunk 3]
```diff
@@ -1,15 +0,0 @@
...
-@@ -62,9 +62,8 @@ constexpr tx_stat parse_status(std::string_view text) noexcept
-   case 'i':
-     if (text == in_progress)
-       PQXX_LIKELY return tx_in_progress;
-     break;
-   }
--  PQXX_UNLIKELY
-   return tx_unknown;
- }
-
```
Comment (FrankXie05) [inline ports/libpqxx/fix_build_with_apple_clang_13.patch:15]: The upstream has been merged.

File: ports/libpqxx/fix_build_with_vs2017.patch [hunk 4]
```diff
@@ -1,8 +1,8 @@
-diff --git a/include/pqxx/compiler-public.hxx b/include/pqxx/compiler-public.hxx
-index 5d24c7e..8087191 100644
---- a/include/pqxx/compiler-public.hxx
-+++ b/include/pqxx/compiler-public.hxx
```
Comment (FrankXie05) [inline ports/libpqxx/fix_build_with_vs2017.patch:4]: Upstream moved file location，re-make this patch.
Upstream related commit: https://github.com/jtv/libpqxx/commit/20fc436375617c6549c6678b615c1b6fc768104e

**Review by Adela0814 — APPROVED** [review]

**Comment by FrankXie05** [comment]
Delete the wrong branch

**Pull request closed without merging**